Building with curl >= 7.87.0 fails: error: 'CURLOPT_PROTOCOLS' is deprecated

**Describe the bug**
Building with curl >= 7.87.0 fails.

```
[ 1094s] In file included from /home/abuild/rpmbuild/BUILD/warzone2100/src/urlrequest.h:23,
[ 1094s]                  from /home/abuild/rpmbuild/BUILD/warzone2100/src/urlrequest.cpp:50:
[ 1094s] /home/abuild/rpmbuild/BUILD/warzone2100/src/urlrequest.cpp: In member function 'virtual CURL* URLTransferRequest::createCURLHandle()':
[ 1094s] /home/abuild/rpmbuild/BUILD/warzone2100/src/urlrequest.cpp:535:42: error: 'CURLOPT_PROTOCOLS' is deprecated: since 7.85.0. Use CURLOPT_PROTOCOLS_STR [-Werror=deprecated-declarations]
[ 1094s]   535 |                 curl_easy_setopt(handle, CURLOPT_PROTOCOLS, CURLPROTO_HTTP | CURLPROTO_HTTPS);
[ 1094s]       |                                          ^~~~~~~~~~~~~~~~~
[ 1094s] /usr/include/curl/curl.h:1749:3: note: declared here
[ 1094s]  1749 |   CURLOPTDEPRECATED(CURLOPT_PROTOCOLS, CURLOPTTYPE_LONG, 181,
[ 1094s]       |   ^~~~~~~~~~~~~~~~~
[ 1094s] cc1plus: all warnings being treated as errors
```

**To Reproduce**
Steps to reproduce the behavior:
Build with curl >= 7.87.0 fails.
Building with curl < 7.87 is fine.

**Expected behavior**
Building with curl >= 7.87 should be fine.

**Your System:**
openSUSE Buildsystem (OBS).
Building for openSUSE Tumbleweed with curl 7.87.
Complete Buildlog: https://build.opensuse.org/build/games/openSUSE_Tumbleweed/x86_64/warzone2100/_log

** Update **
Problem starts with curl 7.87. Building with curl 7.86 is still fine.

--- a/src/init.cpp
+++ b/src/init.cpp
@@ -404,7 +404,7 @@ bool rebuildSearchPath(searchPathMode mode, bool force, const char *current_map,
 			for (const auto& curSearchPath : searchPathRegistry)
 			{
 #ifdef DEBUG
-				debug(LOG_WZ, "Removing [%s] from search path", curSearchPath->path);
+				debug(LOG_WZ, "Removing [%s] from search path", curSearchPath->path.c_str());
 #endif // DEBUG
 				// Remove maps and mods
 				removeSubdirs(curSearchPath->path.c_str(), "maps");
@@ -453,7 +453,7 @@ bool rebuildSearchPath(searchPathMode mode, bool force, const char *current_map,
 			for (const auto& curSearchPath : searchPathRegistry)
 			{
 #ifdef DEBUG
-				debug(LOG_WZ, "Adding [%s] to search path", curSearchPath->path);
+				debug(LOG_WZ, "Adding [%s] to search path", curSearchPath->path.c_str());
 #endif // DEBUG
 				// Add global and campaign mods
 				PHYSFS_mount(curSearchPath->path.c_str(), NULL, PHYSFS_APPEND);
@@ -513,7 +513,7 @@ bool rebuildSearchPath(searchPathMode mode, bool force, const char *current_map,
 			for (const auto& curSearchPath : searchPathRegistry)
 			{
 #ifdef DEBUG
-				debug(LOG_WZ, "Adding [%s] to search path", curSearchPath->path);
+				debug(LOG_WZ, "Adding [%s] to search path", curSearchPath->path.c_str());
 #endif // DEBUG
 				// Add global and multiplay mods
 				PHYSFS_mount(curSearchPath->path.c_str(), NULL, PHYSFS_APPEND);

--- a/src/urlrequest.cpp
+++ b/src/urlrequest.cpp
@@ -530,7 +530,10 @@ public:
 		}
 		curl_easy_setopt(handle, CURLOPT_HEADERFUNCTION, header_callback);
 		curl_easy_setopt(handle, CURLOPT_HEADERDATA, (void *)this);
-	#if LIBCURL_VERSION_NUM >= 0x071304	// cURL 7.19.4+
+	#if LIBCURL_VERSION_NUM >= 0x075500		// cURL 7.85.0+
+		/* only allow HTTP and HTTPS */
+		curl_easy_setopt(handle, CURLOPT_PROTOCOLS_STR, "http,https");
+	#elif LIBCURL_VERSION_NUM >= 0x071304	// cURL 7.19.4+
 		/* only allow HTTP and HTTPS */
 		curl_easy_setopt(handle, CURLOPT_PROTOCOLS, CURLPROTO_HTTP | CURLPROTO_HTTPS);
 	#endif

